jerry-core/api/jerry-debugger.c

```diff
@@ -135,6 +136,13 @@ jerry_debugger_wait_and_run_client_source (jerry_value_t *return_value) /**< [ou
           break;
         }
 
+        /* Stop executing the current context. */
+        if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONTEXT_RESET_MODE))
+        {
+          ret_type = JERRY_DEBUGGER_CONTEXT_RESET_RECEIVED;
```

Comment:
I think these flags should be removed (with binary and) after processed. Otherwise they will always trigger the return. Probably true for JERRY_DEBUGGER_CLIENT_NO_SOURCE.

Reply:
I've added the chnages.